### PR TITLE
fix(tests): EOF - Remove duplicate container tests, automatically check for duplicates

### PR DESCRIPTION
**Pull request by marioevz**
### Duplicate checks

`EOFTest` spec base now automatically checks for duplicate containers during the fill process.
The simple approach is to define a global variable that contains all containers filled up until now, and fail the test in the case that the test currently being generated has already been seen.

An exception to this rule is on `EOFStateTest` specs because these tests can generate multiple state tests on the same container, but in the case of a simple EOF test, if the container has already been seen, the test is gracefully skipped.

## Removed tests

All of the following tests were detected as duplicates and only one of them is left now in code.

- tests/prague/eip7692_eof_v1/eip3540_eof_v1/test_container_validation.py::test_version_validation[fork_CancunEIP7692-eof_test-version_1], tests/prague/eip7692_eof_v1/eip3540_eof_v1/test_container_validation.py::test_magic_validation[fork_CancunEIP7692-eof_test-magic_1_0-magic_0_239]: Same when both valid
- tests/prague/eip7692_eof_v1/eip3540_eof_v1/test_container_validation.py::test_valid_containers[fork_CancunEIP7692-eof_test-single_code_section_with_data_section]
- tests/prague/eip7692_eof_v1/eip3540_eof_v1/test_migrated_valid_invalid.py::test_migrated_valid_invalid[fork_CancunEIP7692-eof_test-EOF1I3540_0011_type_section_size_incomplete] -> tests/prague/eip7692_eof_v1/eip3540_eof_v1/test_container_validation.py::test_invalid_containers[fork_CancunEIP7692-eof_test-incomplete_type_section_size]
- tests/prague/eip7692_eof_v1/eip3540_eof_v1/test_migrated_valid_invalid.py::test_migrated_valid_invalid[fork_CancunEIP7692-eof_test-EOF1I3540_0014_total_of_code_sections_incomplete] -> tests/prague/eip7692_eof_v1/eip3540_eof_v1/test_container_validation.py::test_invalid_containers[fork_CancunEIP7692-eof_test-code_section_count_incomplete]
- tests/prague/eip7692_eof_v1/eip3540_eof_v1/test_migrated_valid_invalid.py::test_migrated_valid_invalid[fork_CancunEIP7692-eof_test-EOF1I3540_0016_code_section_size_incomplete] -> tests/prague/eip7692_eof_v1/eip3540_eof_v1/test_container_validation.py::test_invalid_containers[fork_CancunEIP7692-eof_test-code_section_size_incomplete]
- tests/prague/eip7692_eof_v1/eip3540_eof_v1/test_migrated_valid_invalid.py::test_migrated_valid_invalid[fork_CancunEIP7692-eof_test-EOF1I3540_0017_no_data_section_after_code_section_size] -> tests/prague/eip7692_eof_v1/eip3540_eof_v1/test_container_validation.py::test_invalid_containers[fork_CancunEIP7692-eof_test-truncated_header_data_section]
- tests/prague/eip7692_eof_v1/eip3540_eof_v1/test_migrated_valid_invalid.py::test_migrated_valid_invalid[fork_CancunEIP7692-eof_test-EOF1I3540_0018_no_data_size] -> tests/prague/eip7692_eof_v1/eip3540_eof_v1/test_container_validation.py::test_invalid_containers[fork_CancunEIP7692-eof_test-no_data_section_size]
- tests/prague/eip7692_eof_v1/eip3540_eof_v1/test_migrated_valid_invalid.py::test_migrated_valid_invalid[fork_CancunEIP7692-eof_test-EOF1I3540_0019_data_size_incomplete] -> tests/prague/eip7692_eof_v1/eip3540_eof_v1/test_container_validation.py::test_invalid_containers[fork_CancunEIP7692-eof_test-data_section_size_incomplete]
- tests/prague/eip7692_eof_v1/eip3540_eof_v1/test_migrated_valid_invalid.py::test_migrated_valid_invalid[fork_CancunEIP7692-eof_test-EOF1I3540_0001_invalid_no_magic] -> tests/prague/eip7692_eof_v1/eip3540_eof_v1/test_container_validation.py::test_invalid_containers[fork_CancunEIP7692-eof_test-incomplete_magic]
- tests/prague/eip7692_eof_v1/eip3540_eof_v1/test_migrated_valid_invalid.py::test_migrated_valid_invalid[fork_CancunEIP7692-eof_test-EOF1I3540_0005_invalid_no_version] -> tests/prague/eip7692_eof_v1/eip3540_eof_v1/test_container_validation.py::test_invalid_containers[fork_CancunEIP7692-eof_test-no_version]
- tests/prague/eip7692_eof_v1/eip3540_eof_v1/test_section_order.py::test_container_section_order[fork_CancunEIP7692-eof_test-test_position_CasePosition.HEADER-container_position_2], tests/prague/eip7692_eof_v1/eip3540_eof_v1/test_section_order.py::test_container_section_order[fork_CancunEIP7692-eof_test-test_position_CasePosition.BODY-container_position_2], tests/prague/eip7692_eof_v1/eip3540_eof_v1/test_section_order.py::test_container_section_order[fork_CancunEIP7692-eof_test-test_position_CasePosition.BODY_AND_HEADER-container_position_2] (correct container position)
- tests/prague/eip7692_eof_v1/eip3540_eof_v1/test_section_size.py::test_section_size[fork_CancunEIP7692-eof_test-section_kind_DATA-section_size_NORMAL-exception_None], tests/prague/eip7692_eof_v1/eip3540_eof_v1/test_section_size.py::test_section_size[fork_CancunEIP7692-eof_test-section_kind_TYPE-section_size_NORMAL-exception_None], tests/prague/eip7692_eof_v1/eip3540_eof_v1/test_section_size.py::test_section_size[fork_CancunEIP7692-eof_test-section_kind_CONTAINER-section_size_NORMAL-exception_None] -> tests/prague/eip7692_eof_v1/eip3540_eof_v1/test_section_size.py::test_section_size[fork_CancunEIP7692-eof_test-section_kind_CODE-section_size_NORMAL-exception_None]
- tests/prague/eip7692_eof_v1/eip3540_eof_v1/test_migrated_valid_invalid.py::test_migrated_valid_invalid[fork_CancunEIP7692-eof_test-EOF1V3540_0004_data_section_contents_incomplete] -> tests/prague/eip7692_eof_v1/eip3540_eof_v1/test_section_size.py::test_truncated_container_with_data[fork_CancunEIP7692-eof_test-EOF1_truncated_section_4]
- tests/prague/eip7692_eof_v1/eip3540_eof_v1/test_migrated_valid_invalid.py::test_migrated_valid_invalid[fork_CancunEIP7692-eof_test-EOF1V3540_0003_no_data_section_contents] -> tests/prague/eip7692_eof_v1/eip3540_eof_v1/test_section_size.py::test_truncated_container_with_data[fork_CancunEIP7692-eof_test-EOF1_truncated_section_3]
- tests/prague/eip7692_eof_v1/eip4200_relative_jumps/test_rjump.py::test_rjump_positive_negative_with_data[fork_CancunEIP7692-eof_test] -> tests/prague/eip7692_eof_v1/eip4200_relative_jumps/test_rjump.py::test_rjump_backwards_infinite_loop[fork_CancunEIP7692-eof_test]
- tests/prague/eip7692_eof_v1/eip4750_functions/test_code_validation.py::test_eof_validity[fork_CancunEIP7692-eof_test-callf_to_non_returning] -> tests/prague/eip7692_eof_v1/eip6206_jumpf/test_nonretruning_validation.py::test_returning_section_not_returning[fork_CancunEIP7692-eof_test-stop0]
- tests/prague/eip7692_eof_v1/eip3540_eof_v1/test_container_validation.py::test_invalid_containers[fork_CancunEIP7692-eof_test-code_section_output_too_large_2] -> tests/prague/eip7692_eof_v1/eip6206_jumpf/test_nonretruning_validation.py::test_retf_in_nonreturning[fork_CancunEIP7692-eof_test-PUSH0x0x80-first_False]

## Fixed Tests

The following tests were detected as duplicates but due to a parametrization error:

- tests/prague/eip7692_eof_v1/eip4200_relative_jumps/test_rjumpi.py::test_rjumpi_into_push_1[fork_CancunEIP7692-eof_test-jump_JumpDirection.FORWARD]
- tests/prague/eip7692_eof_v1/eip4200_relative_jumps/test_rjumpv.py::test_rjumpv_into_exchange

Tests are now correctly generating different containers.

## 🔗 Related Issues
None

## ✅ Checklist
- [x] All: Set appropriate labels for the changes.
- [x] All: Considered squashing commits to improve commit history.
- [x] All: Added an entry to [CHANGELOG.md](/ethereum/execution-spec-tests/blob/main/docs/CHANGELOG.md).
- [x] All: Considered updating the online docs in the [./docs/](/ethereum/execution-spec-tests/blob/main/docs/) directory.
- [x] Tests: All converted JSON/YML tests from [ethereum/tests](/ethereum/tests) have been added to [converted-ethereum-tests.txt](/ethereum/execution-spec-tests/blob/main/converted-ethereum-tests.txt).
- [x] Tests: A PR with removal of converted JSON/YML tests from [ethereum/tests](/ethereum/tests) have been opened.
- [x] Tests: Included the type and version of evm t8n tool used to locally execute test cases:  e.g., ref with commit hash or geth 1.13.1-stable-3f40e65.
- [x] Tests: Ran `mkdocs serve` locally and verified the auto-generated docs for new tests in the [Test Case Reference](https://ethereum.github.io/execution-spec-tests/main/tests/) are correctly formatted.

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -37,6 +37,7 @@ dependencies = [
     "rich>=13.7.0,<14",
     "solc-select>=1.0.4,<2",
     "filelock>=3.15.1,<4",
+    "pytest-json-report>=1.5.0,<2",
```
Comment (spencer-tb) [inline pyproject.toml:40]: Just wondering where this is used?

**Review by spencer-tb — APPROVED** [review]
LGTM!

File: pyproject.toml (hunk 1 above)
Comment (marioevz) [reply pyproject.toml:40]: It's not used for this specific PR but I used to get the JSON output from the fill and make it easier for me to detect the failing duplicate containers (it enables the `--json-report` `--json-report-file` flags).

Comment (marioevz) [reply pyproject.toml]: Removed the commit :+1:

**Pull request merged**